### PR TITLE
Add Document.SitePath

### PR DESCRIPTION
**Pull request by yufeih**
`Document.OutputPath` should only be used to indicate where the output file is placed. Use `SitePath` to calculate relative path.

`OutputPath` may contain locale, moniker and other properties.

### CODE REVIEW

File: docs/specs/build/config/basics.yml [hunk 1]
```diff
@@ -147,7 +147,7 @@ inputs:
   docs/a.md:
   docs/b/b.md:
 outputs:
-  output/a.json:
+  output.json:
```
Comment (superyyrrzz) [inline docs/specs/build/config/basics.yml]: tend to make the value only contains only output folder. URL is confusing when considering pretty URL or extension.

File: docs/specs/build/config/basics.yml [hunk 2]
```diff
@@ -158,7 +158,7 @@ inputs:
       docs/a/b.md: outputB
   docs/a/b.md:
 outputs:
-  outputB/b.json:
+  outputB.json:
```
Comment (OsmondJiang) [inline docs/specs/build/config/basics.yml]: >outputB.json: [](start = 2, length = 13)

seems strange?

File: src/docfx/docset/Document.cs [hunk 3]
```diff
@@ -154,22 +159,9 @@ internal static ContentType GetContentType(string path, string docsetPath)
...
-                default:
-                    return path;
-            }
-        }
-
-        internal static string GetSiteUrl(string path, ContentType contentType)
+        internal static string GetSiteUrl(string path, ContentType contentType, Config config)
         {
-            path = '/' + path;
+            path = '/' + ApplyRoutes(path, config.Routes);
```
Comment (OsmondJiang) [inline src/docfx/docset/Document.cs]: >path = '/' + ApplyRoutes(path, config.Routes); [](start = 12, length = 46)

there would be a problem when calculating the relative site url in Resolve

Comment (yufeih) [reply src/docfx/docset/Document.cs]: we'll handle it in url resolve.

**Pull request merged**